vmdb/app/models/metric/ci_mixin/rollup.rb

```diff
@@ -1,12 +1,15 @@
 module Metric::CiMixin::Rollup
-  def perf_rollup_to_parent(interval_name, start_time, end_time = nil)
-    parents = case interval_name
-    when 'realtime'             then [self.perf_rollup_parent(interval_name), interval_name, self, 'hourly']
-    when 'hourly', 'historical' then [self.perf_rollup_parent('hourly'),      interval_name, self, 'daily']
-    when 'daily'                then []
+  def perf_rollup_to_parents(interval_name, start_time, end_time = nil)
+    parent_rollups, next_rollup_interval = case interval_name
+    when 'realtime'             then [self.perf_rollup_parents(interval_name), 'hourly']
+    when 'hourly', 'historical' then [self.perf_rollup_parents('hourly'), 'daily']
+    when 'daily'                then [nil, nil]
     else raise ArgumentError, "invalid interval name #{interval_name}"
     end
 
+    parents = (parent_rollups || []).map { |p| [p, interval_name] }.flatten
+    parents += [self, next_rollup_interval] unless next_rollup_interval.nil?
```

Comment:
@Fryguy `<<` and `+=` don't have the same semantic:

    ([1] << [2,3]) != ([1] + [2,3])
    => true

Here `+=` is the correct one.